src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -545,6 +547,13 @@ rpm_ostree_compose_context_new (const char    *treefile_pathstr,
 {
   g_autoptr(RpmOstreeTreeComposeContext) self = g_new0 (RpmOstreeTreeComposeContext, 1);
 
+  if (!opt_print_only)
+    g_print ("Currently running in %s mode\n",
+             opt_unified_core ? "unified core" : "legacy");
+
+  if (!opt_print_only && !opt_unified_core && !opt_legacy)
+    g_printerr ("%swarning: In the future, the default compose mode will be --unified-mode. The legacy mode will be deprecated but available behind --legacy for some time. Use one of the two switches above to squash this warning.%s\n", get_bold_start (), get_bold_end ());
```

Comment:
We should probably throw a link in there with more details about the difference between the two.